integrationtests/src/test/java/org/axonframework/integrationtests/queryhandling/QueryThreadingIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+                .build();
+        configuration.setCommandThreads(5);
+        configuration.setQueryThreads(5);
+        configuration.setQueryResponseThreads(5);
+        connectionManager = AxonServerConnectionManager.builder()
+                .axonServerConfiguration(configuration)
+                .channelCustomizer(ManagedChannelBuilder::directExecutor)
+                .build();
+        connectionManager.start();
+
+
+        QueryBus localQueryBus = SimpleQueryBus.builder().build();
+        queryBus = AxonServerQueryBus.builder()
+                .axonServerConnectionManager(connectionManager)
+                .configuration(configuration)
+                .localSegment(localQueryBus)
+                .updateEmitter(localQueryBus.queryUpdateEmitter())
+                .messageSerializer(serializer)
+                .genericSerializer(serializer)
+                .build();
+        queryBus.start();
+
+        queryBus.subscribe("query-b", String.class, query -> {
+            while (secondaryQueryBlock.get()) {
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+            return "b";
+        });
+
+        queryBus.subscribe("query-a", String.class, query -> {
+            QueryResponseMessage<String> b = queryBus.query(new GenericQueryMessage<>("start", "query-b", ResponseTypes.instanceOf(String.class))).get();
+            return "a" + b.getPayload();
+        });
+    }
+
+    @AfterEach
+    void tearDown() {
+        commandBus.shutdownDispatching();
+        queryBus.shutdownDispatching();
+
+        commandBus.disconnect();
+        queryBus.disconnect();
+
+        connectionManager.shutdown();
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void canStillHandleQueryResponsesWhileManyQueriesHandling() throws InterruptedException {
+
+        CompletableFuture<QueryResponseMessage<String>> query1 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query2 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query3 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query4 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query5 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+        CompletableFuture<QueryResponseMessage<String>> query6 = queryBus.query(new GenericQueryMessage<>("start", "query-a", ResponseTypes.instanceOf(String.class)));
+
+        Thread.sleep(500);
```

Comment:
Can't you use `await()` for this too, to validate something's false of the upcoming 500ms?

Reply:
Yes, done :)